versions/development/SPEC.md

```diff
@@ -1218,81 +1270,314 @@ $runtime = 'runtime' $ws* '{' ($ws* $runtime_kv $ws*)* '}'
+
+* `"*"` - This special value indicates that ALL returnCodes should be considered a success
+* `Int` - Only the specified return code should be considered a success
+* `Array[Int]` - Only the return codes specify in the array should be considered a success
+
+
+**Single return code**
+```wdl
+task maxRetries_test {
+  #.....
+  runtime {
+    returnCodes: 1
+  }
+}
+```
+
+**Multiple return codes**
+```wdl
+task maxRetries_test {
+  #.....
+  runtime {
+    returnCodes: [1,2,5,10]
+  }
+}
+```
+
+**All return codes**
+```wdl
+task maxRetries_test {
+  #.....
+  runtime {
+    returnCodes: "*"
+  }
+}
+```
+
+### Hints Section
+
+```
+$hints = 'hints' $ws* '{' ($ws* $hints_kv $ws*)* '}'
+$hints_kv = $identifier $ws* ':' $ws* $hints_value
+$hints_value = $string | $number | $boolean | 'null' | $hints_object | $hints_array
+$hints_object = '{}' | '{' $hints_kv (, $hints_kv)* '}'
+$hints_array = '[]' |  '[' $hints_value (, $hints_value)* ']'
+```
+
+The hints section is purely optional and contains key/value pairs. The purpose of this section is to enable the author a means to encode information directly into the task which can be used by an execution engine, which understands the provided information, to optimize the execution of a task. An important distinction to make between `runtime` and `hints` is that a task cannot run without the parameters defined within the `runtime` section, however all properties within the `hints` section are optional and a task SHOULD be able to execute even if the engine completely ignored the entire hints section. Therefore, if there are specific  values within a `hints` section which an engine is expecting, then the engine must always define reasonable default values and not have a strict requirement for the author to provide them.
+
+It must also be clarified that the purpose of this section is **NOT** to provide an execution engine a means to encode all sorts of required values that it needs in order to execute a task without setting any default values. This practice should strictly be considered an anti-pattern and should be avoided by execution engines
+
+The values within the `hints` section are arbitrary K/V pairs, and it is generally up to the execution engine to define what values they are interested in. The provided values can be simple (`String`,`boolean`,`number`...)  or more complex (`Arrays`,`hints object`)
+
+```wdl
+task foo {
+  ...
+  hints {
+		
+  }
+}
+```
+
+#### Reserved Keys
+
+Although `hints` are arbitrary, there are a number of keys which are reserved by the language in or oder to try and encourage interoperability of tasks and workflows between different execution engines. The list of Reserved keys is likely to grow, and in order to prevent name space collisions, engines should follow the [best practices](#conventions-and-best-practices) for specifying hints. 
```

Comment:
Typo `or oder`

Reply:
Fixed a few other typos.. You are eagle eyed